modules/contacts/client/controllers/add-contact.client.controller.js

```diff
@@ -68,7 +68,12 @@
         vm.success = 'Done! We sent an email to your contact and he/she still needs to confirm it.';
       }, function(error) {
         vm.isLoading = false;
-        vm.error = error.message || 'Something went wrong. Try again.';
+        if (error.status === 409) {
+          // 409 means contact already existed
+          vm.success = (error.confirmed) ? 'You two are already connected. Great!' : 'Connection already initiated; now it has to be confirmed.';
```

Comment:
I thought showing it as a success is appropriate. It is then in the same state as if they had just visited that page and the contact already existed. There is nothing the users needs to do too, so an error seems harsh.